website/www/site/content/en/blog/deploy-python-pipeline-on-flink-runner.md

```diff
@@ -0,0 +1,857 @@
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Deploy Python pipelines on Kubernetes using the Flink runner
+
+The [Apache Flink Kubernetes Operator](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/concepts/overview/) acts as a control plane to manage the complete deployment lifecycle of Apache Flink applications. With the operator, we can simplify the deployment and management of Apache Beam pipelines.
+
+In this post, we develop an [Apache Beam](https://beam.apache.org/) pipeline using the [Python SDK](https://beam.apache.org/documentation/sdks/python/) and deploy it on an [Apache Flink](https://flink.apache.org/) cluster by using the [Apache Flink runner](https://beam.apache.org/documentation/runners/flink/). We first deploy an [Apache Kafka](https://kafka.apache.org/) cluster on a [minikube](https://minikube.sigs.k8s.io/docs/) cluster, because the pipeline uses Kafka topics for its data source and sink. Then, we develop the pipeline as a Python package and add the package to a custom Docker image so that Python user code can be executed externally. For deployment, we create a Flink session cluster using the [Flink Kubernetes Operator](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/), and deploy the pipeline using a Kubernetes job. Finally, we check the output of the application by sending messages to the input Kafka topic using a Python producer application.
+
+{{< toc >}}
+
+<!--more-->
+
+## Resources to run a Python Beam pipeline on Flink
+
+We develop an Apache Beam pipeline using the Python SDK and deploy it on an Apache Flink cluster using the Apache Flink runner. While the Flink cluster is created by the Flink Kubernetes Operator, we need two components to run the pipeline on the *Flink runner*: the **job service** and the [**SDK harness**](https://beam.apache.org/documentation/runtime/sdk-harness-config/). Roughly speaking, the job service converts details about a Python pipeline into a format that the Flink runner can understand. The SDK harness executes the Python user code. The Python SDK provides convenience wrappers to manage those components, and it can be utilized by specifying *FlinkRunner* in the pipeline option, for example, `--runner=FlinkRunner`. The *job service* is managed automatically. We rely on our own *SDK harness* as a sidecar container for simplicity. Also, we need the **Java IO Expansion Service**, because the pipeline uses Apache Kafka topics for its data source and sink, and the *Kafka Connector I/O* is developed in Java. Simply put, the expansion service is used to serialize data for the Java SDK.
```

Comment:
```suggestion
We develop an Apache Beam pipeline using the Python SDK and deploy it on an Apache Flink cluster using the Apache Flink runner. Although the Flink cluster is created by the Flink Kubernetes Operator, we need two components to run the pipeline on the *Flink runner*: the **job service** and the [**SDK harness**](https://beam.apache.org/documentation/runtime/sdk-harness-config/). Roughly speaking, the job service converts details about a Python pipeline into a format that the Flink runner can understand. The SDK harness executes the Python user code. The Python SDK provides convenience wrappers to manage those components, and you can use it by specifying *FlinkRunner* in the pipeline option, for example, `--runner=FlinkRunner`. The *job service* is managed automatically. We rely on our own *SDK harness* as a sidecar container for simplicity. Also, we need the **Java IO Expansion Service**, because the pipeline uses Apache Kafka topics for its data source and sink, and the Kafka Connector I/O is developed in Java. Simply put, the expansion service is used to serialize data for the Java SDK.
```